src/metabase/query_processor/middleware/parameters/mbql.clj

```diff
@@ -117,5 +117,6 @@
 
       :else
       (let [filter-clause (build-filter-clause query (assoc param :value param-value))
-            query         (mbql.u/add-filter-clause query filter-clause)]
+            [_ _ opts]    target
+            query         (mbql.u/add-filter-clause query (:stage-number opts) filter-clause)]
```

Comment:
I think seeing a full test in `qp/parameters` with the expected parameter with `:stage-number` in opts would be useful to see how it's all expected to work together.